### PR TITLE
182 add tests for callbacks

### PR DESCRIPTION
**Pull request by lojzatran**
I spent some time to add tests for warning and error callbacks in custom sync.

This PR should increase code coverage.

### CODE REVIEW

File: src/test/java/com/commercetools/project/sync/customer/CustomerSyncerTest.java [hunk 1]
```diff
@@ -55,4 +71,67 @@ void getQuery_ShouldBuildCustomerQuery() {
...
+        .thenReturn(CompletableFuture.completedFuture(pagedQueryResult));
+
+    // test
+    final CustomerSyncer customerSyncer =
+        CustomerSyncer.of(sourceClient, targetClient, mock(Clock.class));
+
+    // assertion
+    customerSyncer.sync(null, true).toCompletableFuture().join();
+    final LoggingEvent errorLog = syncerTestLogger.getAllLoggingEvents().get(0);
+    assertThat(errorLog.getMessage())
```
Comment (ahmetoz) [inline src/test/java/com/commercetools/project/sync/customer/CustomerSyncerTest.java:97]: how does it look like the error message coming from an error callback?

**Review by ahmetoz — COMMENTED** [review]
Looks good, Are you planning to add similar test cases for other resources? Or with a different PR?

File: src/test/java/com/commercetools/project/sync/customer/CustomerSyncerTest.java (hunk 1 above)
Comment (lojzatran) [reply src/test/java/com/commercetools/project/sync/customer/CustomerSyncerTest.java:97]: Do you mean this? https://github.com/commercetools/commercetools-project-sync/pull/193/files/a3322dcf32d0e217e753b5249b6c2cae851dd9de#diff-b80b4644f425dc2d5c9a91c50d59f6a22475b1f5aed3a7e23f96a1ca74cdf8ceR99

**Comment by lojzatran** [comment]
> Looks good, Are you planning to add similar test cases for other resources? Or with a different PR?

Yes, I think I will add some of them here. Depends how much time is left. It's not high prio task for other resources, so I will try to add them only when there's no other tasks to do.

File: src/test/java/com/commercetools/project/sync/customer/CustomerSyncerTest.java (hunk 1 above)
Comment (ahmetoz) [reply src/test/java/com/commercetools/project/sync/customer/CustomerSyncerTest.java:97]: No, an error message coming with `commercetools-sync-java` library.

Comment (lojzatran) [reply src/test/java/com/commercetools/project/sync/customer/CustomerSyncerTest.java:97]: https://github.com/commercetools/commercetools-project-sync/pull/193/commits/da1b928bdb2dfd9f9256c449a0739dff705b93fe

**Review by ahmetoz — APPROVED** [review]
good improvements in here 👍🏽

**Pull request merged**